Sources/HummingbirdBasicAuth/BasicAuthenticator.swift

```diff
@@ -37,7 +38,7 @@ public struct BasicAuthenticator<Context: AuthRequestContext, Repository: Passwo
     ///   - getUser: Closure returning user type
     public init<User: BasicAuthenticatorUser>(
         passwordVerifier: Verifier = BcryptPasswordVerifier(),
-        getUser: @escaping @Sendable (String) async throws -> User?
+        getUser: @escaping @Sendable (String, Logger) async throws -> User?
```

Comment:
Why do we need the logger? Can we wrap this in a new Context type so we can add more metadata in the future as well?

Reply:
The logger is needed because you will most likely be getting the user from a database and the database call will need a logger.

Reply:
I've actually implemented that in #55. If you are happy with #55 then we can close this

Reply:
Since #55 is closed, can we implement that change here?

Reply:
Its reopened as #56